src/main/java/com/puppycrawl/tools/checkstyle/checks/sizes/RecordComponentNumberCheck.java

```diff
@@ -0,0 +1,230 @@
+ * &lt;/module&gt;
+ * </pre>
+ * <p>
+ * Java code example:
+ * </p>
+ * <pre>
+ * public record MyRecord1(int x, int y, String str) { // ok, public record definition allowed 10
+ *     ...
+ * }
+ *
+ * private record MyRecord2(int x, int y, String str, Node node) { // violation
+ *     ...                                // private record definition allowed 3 components
+ * }
+ * </pre>
+ * <p>
+ * Parent is {@code com.puppycrawl.tools.checkstyle.TreeWalker}
+ * </p>
+ * <p>
+ * Violation Message Keys:
+ * </p>
+ * <ul>
+ * <li>
+ * {@code too.many.components}
+ * </li>
+ * </ul>
+ *
+ * @since 8.36
+ */
+@FileStatefulCheck
+public class RecordComponentNumberCheck extends AbstractCheck {
+
+    /**
+     * A key is pointing to the warning message text in "messages.properties"
+     * file.
+     */
+    public static final String MSG_KEY = "too.many.components";
+
+    /** Default maximum number of allowed components. */
+    private static final int DEFAULT_MAX_COMPONENTS = 8;
+
+    /** Specify the maximum number of components allowed in record definition headers. */
+    private int max = DEFAULT_MAX_COMPONENTS;
+
+    /**
+     * Access modifiers of record definitions where the number
+     * of record components should be checked.
+     */
+    private AccessModifierOption[] accessModifiers = {
+        AccessModifierOption.PUBLIC,
+        AccessModifierOption.PROTECTED,
+        AccessModifierOption.PACKAGE,
+        AccessModifierOption.PRIVATE,
+    };
+
+    /**
+     * Setter to specify the maximum number of components allowed in record definition headers.
+     *
+     * @param value the maximum allowed.
+     */
+    public void setMax(int value) {
+        max = value;
+    }
+
+    /**
+     * Setter to access modifiers of record definitions where the number of record
+     * components should be checked.
+     *
+     * @param accessModifiers access modifiers of record definitions which should be checked.
+     */
+    public void setAccessModifiers(AccessModifierOption... accessModifiers) {
+        this.accessModifiers =
+                Arrays.copyOf(accessModifiers, accessModifiers.length);
+    }
+
+    @Override
+    public int[] getDefaultTokens() {
+        return getAcceptableTokens();
+    }
+
+    @Override
+    public int[] getAcceptableTokens() {
+        return new int[] {
+            TokenTypes.RECORD_DEF,
+        };
+    }
+
+    @Override
+    public int[] getRequiredTokens() {
+        return new int[] {
+            TokenTypes.RECORD_DEF,
+        };
+    }
+
+    @Override
+    public void visitToken(DetailAST ast) {
+        // Check access modifier of record definition
+        final DetailAST modifiers =
+            ast.findFirstToken(TokenTypes.MODIFIERS);
+        final AccessModifierOption accessModifier =
+            CheckUtil.getAccessModifierFromModifiersToken(modifiers);
+
+        // If access modifier of record definition matches access modifier options,
+        // count components
```

Comment:
remove comment, code is self descriptive

Reply:
Removed.